vnet/lib/vnet/openflow/ovs_ofctl.rb

```diff
@@ -17,17 +17,30 @@ def initialize(datapath_id)
       @ovs_ofctl = 'ovs-ofctl -O OpenFlow13'
       @ovs_ofctl_10 = 'ovs-ofctl -O OpenFlow10'
       @ovs_vsctl = 'ovs-vsctl'
-      @ovs_vsctl += " --db=#{conf.ovsdb}" if conf.ovsdb
+
+      if conf.ovsdb
+        `ovsdb-client list-dbs #{conf.ovsdb}`
```

Comment:
These two checks to see if the ovsdb and switch are reachable really should be put in separate methods, with the raising of the error where it is now. Not really that nice to read inline system calls in the middle of an initialize method.